Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -137,7 +290,225 @@ public void RoslynConfig_ValidProfile()
+            {"sonaranalyzer.security.cs.ruleNamespace", "SonarAnalyzer.Security"},
+            {"sonaranalyzer.security.cs.pluginKey", "OLD securitycsharpfrontend"},
+            {"sonaranalyzer.security.cs.pluginVersion", "OLD 1.13.0"},
+            {"sonaranalyzer.security.cs.staticResourceName", "OLD SecurityAnalyzer.zip"}
+        });
+        var context = new Context(TestContext, sonarProperties, [[@"c:\assembly1.dll"], [@"d:\foo\assembly2.dll"]], [new SonarRule("roslyn.sonaranalyzer.security.cs", "S2076", true)]);
+
+        var expectedSonarLintXml = """
+            <?xml version="1.0" encoding="UTF-8"?>
+            <AnalysisInput>
+              <Settings>
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.pluginKey</Key>
+                  <Value>securitycsharpfrontend</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.pluginVersion</Key>
+                  <Value>1.13.0</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.staticResourceName</Key>
+                  <Value>SecurityAnalyzer.zip</Value>
+                </Setting>
+              </Settings>
+              <Rules>
+              </Rules>
+              <Files>
+              </Files>
+            </AnalysisInput>
+
+            """;
+        var expectedSecurityRules = """
+        <?xml version="1.0" encoding="utf-8"?>
+        <RuleSet xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Rules for SonarQube" Description="This rule set was automatically generated from SonarQube" ToolsVersion="14.0">
+          <Rules AnalyzerId="SonarScannerFor.NET" RuleNamespace="SonarScannerFor.NET">
+            <Rule Id="S2076" Action="{0}" />
+          </Rules>
+        </RuleSet>
+        """;
+        context.AssertCorrectAnalyzerSettings();
+        context.AssertNoWarningsOrErrors();
+        context.AssertInfoMessages("Provisioning analyzer assemblies for cs...");
+        context.AssertCorrectRulesets(expectedSecurityRules);
+        context.AssertExpectedAssemblies(@"c:\assembly1.dll", @"d:\foo\assembly2.dll");
+        context.AssertExpectedPluginsRequested(
+            [new() { Key = "securitycsharpfrontend", Version = "1.13.0", StaticResourceName = "SecurityAnalyzer.zip" }]);
+        context.AssertExpectedAdditionalFileExists(expectedSonarLintXml);
+    }
+
+    [TestMethod]
+    public void RoslynConfig_IncompletePluginDoesNotPopulate()
+    {
+        var sonarProperties = new ListPropertiesProvider(new Dictionary<string, string>
+            {
+                {"wintellect.pluginKey", "wintellect"},
+                {"sonar.cs.analyzer.dotnet.analyzerId", "SonarAnalyzer.CSharp"},
+                {"sonar.cs.analyzer.dotnet.ruleNamespace", "SonarAnalyzer.CSharp"},
+                {"sonar.cs.analyzer.dotnet.staticResourceName", "SonarAnalyzer.zip"},
```

Comment:
What is this trying to say? That any missing property will be no-op? It's bit hard to see. We should make that more explicit. Something like this:
```
            {"sonar.cs.analyzer.security.missing-pluginKey", "securitycsharpfrontend"},
            {"sonar.cs.analyzer.security.pluginVersion", "1.13.0"},
            {"sonar.cs.analyzer.security.staticResourceName", "SecurityAnalyzer.zip"},
```
The question is how to make it visible for all 3 of them. Parametrized test with `analyzerIdName, versionName, staticResourceName`?